osf/models/sanctions.py

```diff
@@ -917,7 +917,7 @@ def _on_complete(self, event_data):
 
         self.save()
 
-        doi = registration.request_identifier('doi')
+        doi = registration.request_identifier('doi')['doi']
```

Comment:
Should these be `.get('doi', None)`?

Reply:
it should probably wake up Sentry if this fails, but just making the doi None makes it easy to identify times when datacite is down, though it probably fails prior to this.

Reply:
Should it block execution of the rest of this function?

Reply:
Looks like it's guaranteed to, or will break in a currently unhandled way anyway.